tests/timeout_service.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2023 MOSEC Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import multiprocessing
+import random
+import shlex
+import socket
+import subprocess
+import time
+from typing import Any, Type
+
+import httpx
+import pytest
+
+from mosec import Server, Worker
+
+
+class SleepyInference1(Worker):
+    """Sample Class."""
+
+    def forward(self, data: Any) -> Any:
+        time.sleep(1.9)
+        return data
+
+
+class SleepyInference2(Worker):
+    """Sample Class."""
+
+    def forward(self, data: Any) -> Any:
+        time.sleep(2)
+        return data
+
+
+class SleepyInference4(Worker):
+    """Sample Class."""
+
+    def forward(self, data: Any) -> Any:
+        time.sleep(4)
+        return data
```

Comment:
Why not use `def __init__(self, timeout)`?

Reply:
Because `append_worker` needs a worker class parameter rather than object.
I've tried a lot of methods, but none works,
- `server.append_worker(partial(Inference,timeout))` would fail because of `assert issubclass(worker, Worker)`.
-  construct a temperary class with timeout closure, like
```python
def get_cls(timeout):
    class T:
        def forward(self,data):
            time.sleep(timeout)
    return T
ClosureInference=get_cls(timeout)
server.append_worker(ClosureInference)
```
would fail because you can't pass closure to multiprocessing.
- use class static_value, `Inference.timeout=timeout`, in `Inference.forward` run`time.sleep(Inference.timeout)` would also fail, Inference.timeout would'nt change like normal class static value.

Reply:
I see. `os.environ` is all you need. Check the examples.

Reply:
Thx, it has been bothering me all afternoon.